migrations/2018_06_27_100700_change_post_likes_add_foreign_keys.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * (c) Toby Zerner <toby.zerner@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Schema\Builder;
+
+return [
+    'up' => function (Builder $schema) {
+        // Delete rows with non-existent entities so that we will be able to create
+        // foreign keys without any issues.
+        $schema->getConnection()
+            ->table('post_likes')
+            ->whereNotExists(function ($query) {
+                $query->selectRaw(1)->from('posts')->whereColumn('id', 'post_id');
+            })
+            ->orWhereNotExists(function ($query) {
+                $query->selectRaw(1)->from('users')->whereColumn('id', 'user_id');
+            })
+            ->delete();
+
+        $schema->table('post_likes', function (Blueprint $table) {
+            $table->foreign('post_id')->references('id')->on('posts')->onDelete('cascade');
+            $table->foreign('user_id')->references('id')->on('users')->onDelete('cascade');
+        });
```

Comment:
I wonder if we should extract this pattern into some kind of util / helper, to avoid errors?